### ISSUE TITLE
Preparing for Blender 4.0 API breaking changes

### ISSUE THREAD

**Issue by Gorgious56** [IfcOpenShell/IfcOpenShell#3876]
Blender 4.0 is due to release in a few weeks, and since it's a major version change there are a few API breakages to expect. Most of them are documented [in the official wiki](https://wiki.blender.org/wiki/Reference/Release_Notes/4.0/Python_API#:%7E:text=Node%20Groups,to%20NodeTree.interface)

I think it would be a good idea to parse the code base and start versioning code to account for the probable breakages.

At first glance here's some of the changes I think can have an impact :
- changes to the Mesh data structure which might break some of the internal processing.
- deprecation of the old operator override method in favor of `bpy.context.temp_override` - This is a big one
- changes in the nodes API, especially the input/outputs and node groups - only a problem for the section cutaway shader ?
- semi-deprecation of the bgl module (I think this one is aready taken care of)

**Comment by Andrej730** [comment]
I've moved to Blender 4.0 Beta now. 
If someone want to try it out to, to help catch some bugs before the release - it's available on [Blender Daily Builds page](https://builder.blender.org/download/daily/) 

So the old way to override the context is completely deprecated and now is throwing an error like (added it here so it will be possible to find in issues, if someone will try to troubleshoot):
```
Traceback (most recent call last):
  File "<blender_console>", line 1, in <module>
  File "\4.0\scripts\modules\bpy\ops.py", line 106, in __call__
    C_exec, C_undo = _BPyOpsSubModOp._parse_args(args)
  File "\4.0\scripts\modules\bpy\ops.py", line 60, in _parse_args
    raise ValueError("1-2 args execution context is supported")
ValueError: 1-2 args execution context is supported
```

It seems Blender new `temp_override` has an issue with overriding `screen` property ([reported it to Blender](https://projects.blender.org/blender/blender/issues/108763)) which is breaking `bpy.ops.node.clipboard_copy` and `bpy.ops.node.clipboard_paste` we used to copy the shader node trees for the external shaders (shader trees are still not a separate data block and it's possible to copy them by some other way, except recreating them, I guess). I even submitted a [PR to Blender](https://projects.blender.org/blender/blender/pulls/114269), hopefully it will work out.

Some other note interesting thing about context override I've found - `context.object` seems to take priority before `context.active_object` in default blender operators since they are using the method below to determine active object and it's checking `object` first and only then `active_object`, so if you override only `.active_object` it may still use not overriden `.object`.
![image](https://github.com/IfcOpenShell/IfcOpenShell/assets/9417531/e5406c07-7dab-4f05-922e-809fd443aeb8)

**Comment by Gorgious56** [comment]
Wow awesome stuff ! Thanks for the PR, hopefully it will be merged because it looks like an oversight from the devs. I didn't know about the object/active object precedence, I wish this wasn't that foggy, I still don't understand the difference between context.object and context.active_object...

**Comment by Andrej730** [comment]
The [PR fixing `screen` override](https://projects.blender.org/blender/blender/pulls/114269) that fixed External Styles support for Blender 4.0+ was merged into Blender but it will available only in Blender 4.1 release as they don't plan for any more corrective 4.0.x releases atm.

**Comment by Gorgious56** [comment]
Looks like this is resolved, I didn't witness or hear about any bug related to API breakage apart from the section cutaway shader which I fixed recently #4308 . Cheers @Andrej730
